ext/mbstring/mbstring.c

```diff
@@ -2280,8 +2281,8 @@ PHP_FUNCTION(mb_strstr)
 	}
 
 	if (needle.len == 0) {
-		php_error_docref(NULL, E_WARNING, "Empty delimiter");
-		RETURN_FALSE;
+		zend_value_error("Empty delimiter");
+		return;
```

Comment:
I believe we made empty needles legal in str* functions, but didn't follow up with the same change in mbstring. It would be good to do that (as a separate PR...)